tools/cldr-code/src/main/java/org/unicode/cldr/util/CLDRFile.java

```diff
@@ -632,6 +647,51 @@ public CLDRFile add(String currentFullXPath, String value) {
         return this;
     }
 
+    /**
+     * Get the possibly modified value. If this file is resolving, and value matches the bailey value
+     * or inheritance marker, possibly change it from bailey value to inheritance marker, or vice-versa,
+     * as needed to meet these requirements:
+     * 1. If the path changes when getting bailey, then we are inheriting sideways. We need to use a hard value.
+     * 2. If the value is different from the bailey value, can't use inheritance; we need a hard value.
+     * 3. Otherwise we use inheritance marker.
+     *
+     * @param path the path
+     * @param value the input value
+     * @return the possibly modified value
+     */
+    public String reviseInheritanceAsNeeded(String path, String value) {
+        if (isResolved()) {
```

Comment:
If someone calls this on a non-resolved cldrfile, it should throw; otherwise people will think it is producing a valid answer when it isn't.

Reply:
OK, in that case the caller CLDRFile.add needs to check isResolved before calling reviseInheritanceAsNeeded -- since CLDRFile.add does get sometimes get called with isResolved false